### PR TITLE
feat(ref): add kms ref/selector in replication configuration

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
add kms ref/selector in replication configuration for `aws_s3_bucket_replication_configuration`

Fixes #

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

File: config/s3/config.go
Comment (sergenyalcin) [inline config/s3/config.go:108]: If I understand correctly, we need to capture the ID of kms.Key resource as the reference value. Without overriding the extractor, we will use the default `ExternalName` extractor. What do you think about adding the ID extractor: 

```suggestion
			Type: "github.com/upbound/provider-aws/apis/kms/v1beta1.Key",
			Extractor: common.PathTerraformIDExtractor,
```

**Review by sergenyalcin — COMMENTED** [review]
Thanks @haarchri I left a comment

File: config/s3/config.go
Comment (haarchri) [reply config/s3/config.go:108]: in the field description for replica_kms_key_id i can see 

`	// ID (Key ARN or Alias ARN) of the customer managed AWS KMS key stored in AWS Key Management Service (KMS) for the destination bucket.`

**Comment by haarchri** [comment]
@sergenyalcin i can rebase - can we merge this after ?

File: config/s3/config.go
Comment (sergenyalcin) [reply config/s3/config.go:108]: As I see, the external name of kms.Key is arn of this resource so the default extractor works.

**Review by sergenyalcin — APPROVED** [review]
Thanks @haarchri LGTM! After rebasing we can merge this.

**Comment by turkenf** [comment]
/test-examples="examples/ec2/vpc.yaml"

**Pull request merged**
